util/string_util.cc

```diff
@@ -279,9 +279,16 @@ bool StartsWith(const std::string& string, const std::string& pattern) {
 #ifndef ROCKSDB_LITE
 
 bool ParseBoolean(const std::string& type, const std::string& value) {
-  if (value == "true" || value == "1") {
+  const static std::string kTrue = "true", kFalse = "false";
```

Comment:
Why?

Reply:
There was a confusing usage in the client code (`ParseCompressionOptions()`) where it passed all remaining fields in `value`. It only worked before this PR because the boolean (`CompressionOptions::enabled`) was the final option to parse, so the remaining fields would consist of only that field. This PR exposed the issue by adding an option (`CompressionOptions::max_dict_buffer_bytes`) that should be parsed after the boolean.

Anyways, I fixed the confusing usage in `ParseCompressionOptions()` now (see reply above), and reverted this change.